File: June 2025/12.py
class Solution:
    def printKClosest(self, arr, k, x):
        # Create a list to store the differences and corresponding elements
        diff = []
        
        # Iterate through the array and calculate the difference
        for num in arr:
            if num != x:
                diff.append((abs(num - x), -num))  # Using -num to ensure higher values come first in case of a tie
        
        # Sort the list based on the absolute difference and then by element value in descending order
        diff.sort()
        
        # Extract the k closest elements
        result = [-y[1] for y in diff[:k]]
        
        return result

File: June 2025/12.cpp
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
  public:
    vector<int> printKClosest(vector<int> arr, int k, int x) {
        vector<pair<int, int>> diff; 
        
        // Compute absolute differences while excluding x
        for (int num : arr) {
            if (num != x) {
                diff.push_back({abs(num - x), -num}); // Use -num for prioritizing larger values
            }
        }

        // Sort based on absolute difference, and prioritize larger numbers in case of a tie
        sort(diff.begin(), diff.end());

        vector<int> result;
        for (int i = 0; i < k && i < diff.size(); i++) {
            result.push_back(-diff[i].second); // Restore original number
        }

        return result;
    }
};
